go/interpreter/lfvm/instructions_test.go

```diff
@@ -1758,3 +1751,84 @@ func TestInstructions_ComparisonAndShiftOperations(t *testing.T) {
 		})
 	}
 }
+
+func TestOpBlockhash(t *testing.T) {
+
+	hash := tosca.Hash(uint256.MustFromHex("0xABCD").Bytes32())
+	zeroHash := tosca.Hash{}
+	u64overflow := new(uint256.Int).Add(uint256.NewInt(2), uint256.NewInt(math.MaxUint64))
+
+	type testInput struct {
+		currentBlockNumber   int64
+		requestedBlockNumber *uint256.Int
+	}
+
+	tests := map[string]struct {
+		expectedValue tosca.Hash
+		inputs        map[string]testInput
+	}{
+		"produces zero if requested block number older than available history": {
+			expectedValue: zeroHash,
+			inputs: map[string]testInput{
+				"default": {currentBlockNumber: 1024, requestedBlockNumber: uint256.NewInt(36)},
+			},
+		},
+		"produces zero if requested block number newer than current": {
+			expectedValue: zeroHash,
+			inputs: map[string]testInput{
+				"default": {currentBlockNumber: 500, requestedBlockNumber: uint256.NewInt(501)},
```

Comment:
should this also return zero the requested number is the same as the current?

Reply:
right